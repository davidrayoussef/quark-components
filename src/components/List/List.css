h2 {
  margin: 0;
  font-size: 1rem;
}

ul {
  margin-top: 0;
  padding: 0;
}
