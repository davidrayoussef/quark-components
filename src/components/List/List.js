import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './List.style.css';

const List = ({ title, data }) => {
  return (
    <Fragment>
      <h2>{title}</h2>
      <ul>
        {
          data.map(item => {
            return <li key={item}>{item}</li>
          })
        }
      </ul>
    </Fragment>
  );
};

List.proptypes = {
  title: PropTypes.string,
  data: PropTypes.array.isRequired
};

export default List;
