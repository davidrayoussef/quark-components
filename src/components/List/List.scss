.list-container {
  & .title {
    margin: 0 0 5px;
    font-size: 1rem;
  }
  & .list {
    margin-top: 0;
    padding: 0;
    & li {
      margin-bottom: 5px;
    }
  }
}
