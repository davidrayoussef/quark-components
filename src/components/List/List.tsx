import React from 'react';

import { ListProps } from '@/shared';

import style from './List.scss';

export const List: React.FC<ListProps> = ({ title, data }: ListProps) => {
  return (
    <div className={style.listContainer}>
      <h3 className={style.title}>{title}</h3>
      <ul className={style.list}>
        {data.map((item: string) => {
          return <li key={item}>{item}</li>;
        })}
      </ul>
    </div>
  );
};
