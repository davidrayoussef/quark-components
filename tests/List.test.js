import React from 'react';
import { shallow } from 'enzyme';
import List from '../src/components/List/List';

const data = [
  { name: 'list item 1' },
  { name: 'list item 2' },
  { name: 'list item 3' }
]

test('should render', () => {
  const wrapper = shallow(
    <List data={[]}/>
  );
  expect(wrapper);
});

test('should render a title', () => {
  const wrapper = shallow(
    <List data={[]} listTitle="List Title" />
  );
  expect(wrapper.find('h3').text()).toBe('List Title');
});

test('should render 3 list items', () => {
  const wrapper = shallow(
    <List data={data}/>
  );
  expect(wrapper.find('li').length).toBe(3);
});

test('snapshot', () => {
  const wrapper = shallow(
    <List data={data}/>
  );
  expect(wrapper).toMatchSnapshot();
});
