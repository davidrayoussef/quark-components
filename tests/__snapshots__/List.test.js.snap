// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<React.Fragment>
  <h3
    className="listTitle"
  />
  <ul
    className="list"
  >
    <li
      key="list item 1"
    >
      list item 1
    </li>
    <li
      key="list item 2"
    >
      list item 2
    </li>
    <li
      key="list item 3"
    >
      list item 3
    </li>
  </ul>
</React.Fragment>
`;
